arcgis-ios-sdk-samples/Route & Directions/Find closest facility to multiple incidents (service)/FindClosestFacilityMultipleIncidentsServiceViewController.swift

```diff
@@ -0,0 +1,226 @@
+        
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()
+        dispatchGroup.enter()
+        queryAllFeatures(from: facilitiesLayer.featureTable!) { [weak self] (result) in
+            switch result {
+            case .success(let features):
+                self?.facilityFeatures = features
+            case .failure(let error):
+                self?.presentAlert(error: error)
+            }
+            dispatchGroup.leave()
+        }
+        queryAllFeatures(from: incidentsLayer.featureTable!) { [weak self] (result) in
+            switch result {
+            case .success(let features):
+                self?.incidentFeatures = features
+            case .failure(let error):
+                self?.presentAlert(error: error)
+            }
+            dispatchGroup.leave()
+        }
+        dispatchGroup.notify(queue: .main) { [weak self] in
+            guard let strongSelf = self else { return }
+            
+            let geometries = (strongSelf.facilityFeatures + strongSelf.incidentFeatures).compactMap { $0.geometry }
+            if let extent = AGSGeometryEngine.combineExtents(ofGeometries: geometries) {
+                strongSelf.mapView.setViewpointGeometry(extent, padding: 20)
+            }
+            
+            strongSelf.solveRoutesButtonItem.isEnabled = true
+        }
+        
+        return map
+    }
+    
+    /// Queries for all the features from a given feature table.
+    ///
+    /// - Parameters:
+    ///   - featureTable: The feature table whose features should be queried.
+    ///   - completion: A closure executed upon success or failure.
+    func queryAllFeatures(from featureTable: AGSFeatureTable, completion: @escaping (Result<[AGSFeature], Error>) -> Void) {
+        featureTable.load { [unowned featureTable] (error) in
+            if let error = error {
+                completion(.failure(error))
+            } else {
+                let queryParameters = AGSQueryParameters()
+                queryParameters.whereClause = "1=1"
+                featureTable.queryFeatures(with: queryParameters) { (result, error) in
+                    if let result = result {
+                        completion(.success(result.featureEnumerator().allObjects))
+                    } else if let error = error {
+                        completion(.failure(error))
+                    }
+                }
+            }
+        }
+    }
+    
+    /// Called in response to the Solve Routes button being tapped.
+    @IBAction func solveRoutes() {
+        solveRoutesButtonItem.isEnabled = false
+        closestFacilityTask.defaultClosestFacilityParameters { [weak self] (parameters, error) in
+            if let parameters = parameters {
+                self?.didGetClosestFacilityParameters(parameters)
+            } else if let error = error {
+                print("Error getting default closest facility parameters: \(error)")
```

Comment:
Should we show an alert rather than just printing the error? I see this at other places too.

Reply:
That's a good question and there isn't really a firm answer. As you've seen, throughout all the samples, some places we present an alert, some places we print the error. I suppose it wouldn't hurt to be consistent, at least in this sample.